app/models/concerns/galtersufia/generic_file/header_lookup.rb

```diff
@@ -11,43 +11,39 @@ module HeaderLookup
       END_SPARQL_MESH_URI = "%27%2C%20%27i%27))%20%0D%0A%7D%20%0D%0AORDER%20BY%20%3Fd%20%0D%0A"
       LCSH_BASE_URI = "http://id.loc.gov/authorities/subjects/suggest/?q="
 
-      def pid_lookup_by_scheme(term, scheme)
-        scheme == :mesh ? mesh_term_pid_lookup(term) : lcsh_term_pid_lookup(term)
+      def pid_lookup_by_scheme(term="", scheme="")
+        if term.blank? || scheme.blank?
+          return
+        elsif scheme == :mesh
+          memoized_mesh_lookups[term] || mesh_term_pid_lookup(term)
```

Comment:
if the memoized result is found use it, otherwise return the function's return value.